pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,140 @@
+{ stdenvNoCC, lib, writeTextDir, php, unzip, xz, git }@inputs:
+
+let
+  buildPhpProjectOverride = finalAttrs: previousAttrs:
+
+  let
+    composerLock = finalAttrs.composerLock or null;
+    vendorDir = finalAttrs.vendorDir or "vendor";
+    php = finalAttrs.php or inputs.php;
+
+    composer = if (finalAttrs.pname == "composer") then
```

Comment:
Might be better to just take `composer` in `finalAttrs` inject the `phar` just in the `composer` expression, rather than matching on name instead of (inversion of control). The phar does not even need to be publicly available.

Reply:
Fixed